docs/reference-guides/block-api/block-supports.md

```diff
@@ -114,58 +113,230 @@ supports: {
+When the block declares support for `color.background`, the attributes definition is extended to include two new attributes: `backgroundColor` and `style`:
+
+- `backgroundColor`: attribute of `string` type with no default assigned.
+
+  When a user chooses from the list of preset background colors, the preset slug is stored in the `backgroundColor` attribute.
+
+  Background color presets are sourced from the `editor-color-palette` [theme support](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-color-palettes).
+
+  The block can apply a default preset background color by specifying its own attribute with a default e.g.:
+
+  ```js
+  attributes: {
+      backgroundColor: {
+          type: 'string',
+          default: 'some-preset-background-slug',
+      }
+  }
+  ```
+
+- `style`: attribute of `object` type with no default assigned.
+  
+  When a custom background color is selected (i.e. using the custom color picker), the custom color value is stored in the `style.color.background` attribute.
+
+  The block can apply a default custom background color by specifying its own attribute with a default e.g.:
+
+  ```js
+  attributes: {
+      style: {
+          type: 'object',
+          default: {
+              color: {
+                  background: '#aabbcc',
+              }
+          }
+      }
+  }
+  ```
+
+### color.duotone
+
+This property adds UI controls which allow to apply a duotone filter to a block or part of a block.
 
 ```js
-attributes: {
-    backgroundColor: {
-        type: 'string',
-        default: 'some-value',
+supports: {
+    color: {
+        // Apply the filter to the same selector in both edit and save.
+        duotone: '.duotone-img, .duotone-video',
+
+        // Default values must be disabled if you don't want to use them with duotone.
+        background: false,
+        text: false
     }
 }
 ```
 
-- When `gradients` support is declared: it'll be added a new `gradient` attribute of type `string` with no default assigned. It stores the preset values set by the user. The block can apply a default text color by specifying its own attribute with a default e.g.:
+Duotone presets are sourced from `color.duotone` in [theme.json](https://developer.wordpress.org/block-editor/developers/themes/theme-json/).
+
+When the block declares support for `color.duotone`, the attributes definition is extended to include the attribute `style`:
+
+- `style`: attribute of `object` type with no default assigned.
+
+  The values are split by channel (i.e. 'r' for red, 'g' for green, and 'b' for blue) and represent the intensity of that channel from dark to light using values in the closed interval [0,1].
+
+  The block can apply a default duotone color by specifying its own attribute with a default e.g.:
+
+  ```js
+  attributes: {
+      style: {
+          type: 'object',
+          default: {
+              color: {
+                  duotone: {
+                      values: {
```

Comment:
why do we need a subkey "values" here? do we expect any other subkey?

Reply:
Yeah, right now I have `slug` as another subkey for when a preset is selected. This is what I was asking about in https://github.com/WordPress/gutenberg/pull/26752#discussion_r600775951

Reply:
In other block supports, when we select a "custom" value, we store it in the "style" attribute directly like "values" here but when we select a "preset" we store its "slug" in a dedicated attribute different than style. I think personally it makes sense to store everything like you did in "style" in the future.

That said, I feel like for the moment and since the style is being generated per block, we shouldn't be storing "slug" maybe, because it's just gives the impression that a preset is used (just a class added to the block) while in reality, a style a generated for each usage (even if duplicated sometimes). So I wonder whether it makes sense for v1 to consider all duotone applications as "custom values" and not "presets".

Reply:
Makes sense that it might be confusing to be using the slug differently from how other slugs are being used. I can update the style output to always use the uuid, but the slug is also being used to show the checkbox on the picker (https://github.com/WordPress/gutenberg/pull/26752/files#diff-8b3afbf252bd06e46799a8c6848b9e05625144f3e0264bc675124bcc4eaaa954R34). It seemed complicated to be deeply comparing the value to perform that check.

Reply:
> It seemed complicated to be deeply comparing the value to perform that check.

Yes, that's the downside, how complex is that though? If it's about two colors comparison, I think it's worth a shot.

Reply:
It could, in the future, expand to an arbitrary amount of color stops on the color scale, just like a gradient. The emphasis on 2 has been to focus the user experience and end results. We may not want to go beyond 2, but it'd be nice to be able to.

Reply:
And even now it's more than just a two color comparisons. The swatches only have the array of colors that, on selection, get parsed with tinycolor and converted to the duotone values format. So in order to deeply compare, I also need to convert the color list in the swatch to the duotone values format before comparing.

Reply:
>  get parsed with tinycolor and converted to the duotone values format

This makes me wonder if we should parse before setting the attribute which mean the attribute get saved with the "parsed value" or whether we should do it only when applying the style. In Gutenberg the principle has always been: 

 - In the block JSON (attributes) representation, store the values, the user edits (optimize for edition)
 - When generating HTML and CSS, perform the serialization (compute the values optimized for rendering).

Reply:
In other words, maybe we should store the colors, in the same way colors are stored across Gutenberg?

Reply:
I optimized to avoid re-implementation in JS and PHP. Doing the conversion from string, `#FFAACC` or `hsl(336, 100%, 83%)`, to what I need for the SVG filter is done with `tinycolor` in JS and I wanted to avoid writing parts of tinycolor in PHP or finding an alternative that might not behave exactly the same.